### DISCUSSION TITLE
Problem with HSL formatting

### DISCUSSION THREAD

**Discussion by andreamorales** [coldtype/coldtype#109] [Q&A]
Hi. Bit of a noob here. Why do you define your HSL hue value as a float (example, 0.8), instead of a number from 1-255? Trying to get this blue and can't for the life of me.

<img width="237" alt="Screen Shot 2022-06-27 at 9 54 55 AM" src="https://user-images.githubusercontent.com/21376359/175958593-7b18e02d-9c54-46b2-a551-ca2a5863e971.png">

Thank you!

**Comment by stenson** [comment]
Good question! That is a quirk in Coldtype — the hue is specified as a 0-1 float because 0-1 numbers are idiomatic throughout Coldtype for animation purposes, so you can do things like "rotate" the hue in an `hsl` easily, without needing to convert the animation value returned from `f.e` (which is a number that increases from 0 to 1) from 0-1 to 0-360 for hue. Here's an example of passing an easing value directly to the hsl function.

```python
from coldtype import *

@animation(timeline=60)
def scratch(f):
    return (P(f.a.r)
        .f(hsl(f.e("l", 0))))
```

That said, you could also add a `/360` after a hue value specified in 1-360 format, i.e. for that HSL you've screenshotted above, it should be: `hsl(253/360, .6, .36)`, or like this in a full example:

```python
from coldtype import *

@animation()
def scratch(f):
    return (P(f.a.r)
        .f(hsl(253/360, .6, .36)))
```

---

All _that_ said, I know hsl-in-360 is very common, so coldtype should probably have something like an `hsl360` function that would accept the CSS-style `<0-360>, <0-100>, <0-100>` arguments. I'll make a note to add that for future versions, but if you wanted to add a function like that to your own source code in the meantime, it could look something like this:

```python
from coldtype import *

def hsl360(h, s, l, a=1.0):
    return hsl(h/360, s/100, l/100, a)

@animation()
def scratch(f):
    return (P(f.a.r)
        .f(hsl360(253, 60, 36)))
```
